swarm/api/http/server.go

```diff
@@ -187,6 +187,12 @@ func handler(w http.ResponseWriter, r *http.Request, a *api.Api) {
 			reader := a.Retrieve(key)
 			quitC := make(chan bool)
 			size, err := reader.Size(quitC)
+			if err != nil {
+				glog.V(logger.Debug).Infof("Could not determine size: %v", err.Error())
+				//An error on call to Size means we don't have the root chunk
+				http.Error(w, err.Error(), http.StatusNotFound)
```

Comment:
instead of err.Error() here we should wrap the error in a descriptive context like the one you are logging on line 191